W-14758017 AEH Jan 2024 release jh (#20)
mulesoft/docs-api-experience-hub: modules/ROOT/pages/simulating-api-calls-using-the-mocking-service.adoc

include::_attributes.adoc[]
endif::[]

> Using the mocking service in the API Console of the portal, you can simulate calls to your API. The mocking service is a feature of Anypoint that enables you to generate simulated servers that aid in the development and testing of APIs. These mock servers imitate the actions of accepting requests and providing responses. The mocking service replicates the functionality of a genuine API. 

Reviewer: ```suggestion
To simulate calls to your API, use the mocking service in the API Console of the portal. The mocking service feature of Anypoint enables you to generate simulated servers to help you develop and test APIs. These mock servers imitate the actions of accepting requests and providing responses. The mocking service replicates the functionality of a genuine API. 
```

Reviewer: Suggested way to present - put the user goal first.

== Before You Begin

You must have an API specification with at least one endpoint and one method in it.

== Simulate a Call to an API Using an Internal URL

> To access the internal URL, you must specify the authorization headers as part of the request URL. To test an API without supplying authorization headers, generate a mocking service public URL for this project. Mocking service is the selected server in the *Try it* pane. 

Reviewer 2: It shouldn't be panel instead of pane

Author: Hi Alex, Thanks for pointing that out. Since I'm accustomed to using the word "pane" in the Exchange documentation, I'm also using "pane" consistently in AEH as well.

> To access the internal URL, you must specify the authorization headers as part of the request URL. To test an API without supplying authorization headers, generate a mocking service public URL for this project. Mocking service is the selected server in the *Try it* pane. 

Reviewer: The mocking service?

. On the *Home* or the *APIs* page, select a REST API.
. Select the *API specification* tab.

> . In the *Specification* pane, click a method like `GET`.

Reviewer: Do you need "In the ..." if you've previously told them to select the tab?

. In the *Documentation* pane, review the requests and responses data.

> . In the *Try it* pane, enter and query parameters you want to test. 

Reviewer 2: Idem top

> . In the *Try it* pane, enter and query parameters you want to test. 

Reviewer: ```suggestion
. In the *Try it* pane, enter query parameters to test. 
```

== See Also

> * xref:using-the-api-console.adoc[API Console]

Reviewer: Can the link text be task-oriented?
